### ISSUE TITLE
We should consider completing transformations for ```BoundAttribute``` arguments in "success" mode when the only failure reason is accessibility of attribute's  constructor. 

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#70861]
From ```NullableWalker.cs```:
```
            static void expandParamsArray(ref ImmutableArray<BoundExpression> arguments, ref ImmutableArray<RefKind> refKindsOpt, ImmutableArray<ParameterSymbol> parametersOpt, ref ImmutableArray<int> argsToParamsOpt, ref BitVector defaultArguments)
            {
                // It looks like in some error scenarios we can get here without params array created.
                // At the moment, there is only one test that gets here like that - Microsoft.CodeAnalysis.CSharp.UnitTests.AttributeTests.TestBadParamsCtor.
                // And we get here for the erroneous attribute application, constructor is inaccessible. 
                // Perhaps that shouldn't cancel the default values / params array processing?
                Debug.Assert(arguments.Count(a => a.IsParamsArray) <= 1);
```
